build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,180 @@
+      <PropertyGroup>
+        <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+
+        <ExeSuffix Condition=" '$(OSName)' == 'win' ">.exe</ExeSuffix>
+        <ExeSuffix Condition=" '$(OSName)' != 'win' "></ExeSuffix>
+
+        <DynamicLibPrefix Condition=" '$(OSName)' == 'win' "></DynamicLibPrefix>
+        <DynamicLibPrefix Condition=" '$(OSName)' != 'win' ">lib</DynamicLibPrefix>
+
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'win' ">.dll</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'osx' ">.dylib</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(DynamicLibSuffix)' == '' ">.so</DynamicLibSuffix>
+
+        <DotnetHostBaseName>dotnet$(ExeSuffix)</DotnetHostBaseName>
+        <DotnetHostFxrBaseName>$(DynamicLibPrefix)hostfxr$(DynamicLibSuffix)</DotnetHostFxrBaseName>
+        <HostPolicyBaseName>$(DynamicLibPrefix)hostpolicy$(DynamicLibSuffix)</HostPolicyBaseName>
+
+        <StageDirectory>%(Stage.StageDirectory)</StageDirectory>
+        <StageSymbolsDirectory>%(Stage.StageSymbolsDirectory)</StageSymbolsDirectory>
+        <SdkOutputDirectory>$(StageDirectory)/sdk/$(NugetVersion)</SdkOutputDirectory>            
+        <BinaryToCorehostifyRelDir>runtimes/any/native</BinaryToCorehostifyRelDir>
+        <BinaryToCorehostifyOutDir>$(SdkOutputDirectory)/$(BinaryToCorehostifyRelDir)</BinaryToCorehostifyOutDir>
+        <MSBuildTargetsDirectory>$(SdkOutputDirectory)/runtimes/any/native</MSBuildTargetsDirectory>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <BinObj Include="$(SrcDirectory)/**/bin/**/*" />
+        <BinObj Include="$(SrcDirectory)/**/obj/**/*" />
+        <FilesToRemoveFromStageDirectory Include="$(StageDirectory)/**/*" />
+
+        <SharedFramework Include="$(IntermediateDirectory)/sharedFrameworkPublish/**/*.*" />
+
+        <BinariesToRemove Include="csc" />
+        <BinariesToRemove Include="vbc" />
+
+        <MSBuildTargetsToCopy Include="$(MSBuildTargetsDirectory)/**/*.targets" />
+        <MSBuildTargetsToCopy Include="$(MSBuildTargetsDirectory)/**/*.Targets" />
+        <MSBuildTargetsToCopy Include="$(MSBuildTargetsDirectory)/**/*.props" />
+        <MSBuildTargetsToCopy Include="$(MSBuildTargetsDirectory)/**/*.overridetasks" />
+        <MSBuildTargetsToCopy Include="$(MSBuildTargetsDirectory)/**/*.tasks" />
+      </ItemGroup>
+
+      <SharedFrameworkNameVersionPath RootOutputDirectory="$(StageDirectory)">
+        <Output TaskParameter="OutputSharedFrameworkNameVersionPath" PropertyName="SharedFrameworkNameVersionPath" />
+      </SharedFrameworkNameVersionPath>
+
+      <Delete Files="@(BinObj)" />
+
+      <Delete Files="@(FilesToRemoveFromStageDirectory)" />
+      <RemoveDir Directories="$(StageDirectory)" />
+      <MakeDir Directories="$(StageDirectory)"/>
+
+      <!-- CopySharedFramework -->
+      <Copy SourceFiles="@(SharedFramework)"
+            DestinationFiles="@(SharedFramework->'$(StageDirectory)\%(RecursiveDir)%(Filename)%(Extension)')" />
```

Comment:
`%(RecursiveDir)` is a confusing concept IMO and if the above `SharedFramework` path were to change, this may break. 

Could we use `MakeRelative` here instead?